hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/ContainerEndpoint.java

```diff
@@ -397,7 +419,81 @@ private List<ContainerBlockMetadata> getBlocks(
     return blockIds;
   }
 
-  private BucketLayout getBucketLayout() {
-    return BucketLayout.DEFAULT;
+  /**
+   * Builds an object path for a file system optimized bucket.
+   *
+   * @param prevKeyPrefix the previous key prefix of the object path
+   * @return the object path for the file system optimized bucket
+   * @throws IOException if an IO error occurs
+   */
+  private String buildObjectPathForFileSystemBucket(String prevKeyPrefix)
```

Comment:
I don't see this method being tested anywhere.  Am I missing it.

Also the name sounds awkward because it sounds like it is only return fso paths.  I would probably just call it correctPathForLayout() or something like that.

Reply:
Hi @GeorgeJahad,

I would like to confirm if the current approach for handling FSO paths is correct. Once we reach a consensus, I will proceed with writing the UT for it.